### PR TITLE
Opentelemetry

### PR DESCRIPTION
**Pull request by asonix**
This adds an optional OTLP exporter to lemmy to expose tracing spans as opentelemetry data to be consumed by OTLP-compatible collectors

Additionally, it adds the official opentelemetry collector and the jaeger all-in-one container to the development docker compose to improve visibility in development

No changes were made to the production docker-compose file because I'm not sure a good way to ship an OTLP collector that will work in most environments.

### CODE REVIEW

**Comment by dessalines** [comment]
Looks good to me, and it'll def be useful. I'll wait for nutomic to get back tho before we merge.

**Comment by Nutomic** [comment]
I dont really have time to test this, but if it works for you @dessalines then sure we can merge it. Code seems fine to me.

**Comment by dessalines** [comment]
I got the merge conflicts into this one here: https://github.com/LemmyNet/lemmy/pull/2008

**Review by dessalines — APPROVED** [review]
Seems to work fine, it wasn't clear tho that to access this, you need to go to http://localhost:16686 . 

Might be good to put that as a comment in either of those `yml` files.

**Comment by dessalines** [comment]
Still some merge conflicts, and needs a rebase or merge from main. @Nutomic could you also look this over.

File: config/defaults.hjson
Comment (Nutomic) [inline config/defaults.hjson:120]: Is this supposed to be `#` instead of `//`?

File: crates/websocket/src/handlers.rs
Comment (Nutomic) [inline crates/websocket/src/handlers.rs:73]: Is there any reason you are doing imports inside the function?

File: crates/websocket/src/handlers.rs
Comment (Nutomic) [inline crates/websocket/src/handlers.rs:104]: What a messy code. But it was already like this before...

**Comment by Nutomic** [comment]
Perfect, thanks!

**Pull request merged**
